### PR TITLE
Add host-firewall-egress-to-fqdns test

### PR DESCRIPTION
**Pull request by atykhyy**
This commit adds a connectivity test for host L7 DNS proxy capability added in https://github.com/cilium/cilium/pull/34024.

### CODE REVIEW

**Review by doniacld — APPROVED** [review]
lgtm!

File: connectivity/tests/host.go
Comment (christarazi) [inline connectivity/tests/host.go:235]: Why is this second scenario required?

**Review by christarazi — COMMENTED** [review]
Do we nee to wait for the cilium/cilium PR to be merged first before we merge this one?

**Comment by atykhyy** [comment]
> Do we nee to wait for the cilium/cilium PR to be merged first before we merge this one?

Yes, the test will fail otherwise as the functionality it's testing is not yet present.

File: connectivity/tests/host.go
Comment (atykhyy) [reply connectivity/tests/host.go:235]: I need to apply host policy and test that Cilium both allows host egress to the FQDN specified in policy and denies egress to another FQDN. I looked at analogous pod-level tests in connectivity/builder/to_fqdns.go. These use two scenarios (PodToWorld and PodToWorld2) and I copied that.

**Review by ldelossa — APPROVED** [review]
LGTM. Thanks!

**Comment by atykhyy** [comment]
@ldelossa Please don't merge this yet, the test has problems when bpf-tproxy is enabled. I am working on fixing this.

**Comment by michi-covalent** [comment]
@atykhyy could you move this pull request to cilium/cilium repo? cilium-cli code got moved to https://github.com/cilium/cilium/tree/main/cilium-cli 🙏

**Comment by michi-covalent** [comment]
> I can create a matching pull request there and close this one

yeah this is what i meant 👍

**Pull request closed without merging**
